x-pack/plugins/apm/public/services/rest/apm/apm.ts

```diff
@@ -19,3 +25,54 @@ export async function getEncodedEsQuery(kuery?: string) {
   const esFilterQuery = convertKueryToEsQuery(kuery, indexPattern);
   return encodeURIComponent(JSON.stringify(esFilterQuery));
 }
+
+export async function getKueryESFilter(kuery?: string): Promise<ESFilter | undefined> {
+  if (!kuery) {
+    return;
+  }
+
+  const indexPattern = await getAPMIndexPatternForKuery();
+  if (!indexPattern) {
+    return;
+  }
+
+  return convertKueryToEsQuery(kuery, indexPattern) as ESFilter;
+}
+
+export function getEnvironmentESFilter(environment?: string): ESFilter | undefined {
+
+  if (!environment) {
+    return undefined;
+  }
+
+  if (environment === 'Null') {
+    return {
+      bool: { must_not: { exists: { field: 'service.environment' } } }
+    };
+  }
+  return {
+    term: { 'service.environment': environment }
```

Comment:
`service.environment` should be a constant in https://github.com/elastic/kibana/blob/d50c9702907f755652af2ebff3b3e9730ae49a35/x-pack/plugins/apm/common/elasticsearch_fieldnames.ts